src/NodaTime.Web/Startup.cs

```diff
@@ -80,6 +84,43 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerF
                 ContentTypeProvider = new FileExtensionContentTypeProvider
                 {
                     Mappings = { [".nzd"] = "application/octet-stream" }
+                },
+                OnPrepareResponse = (context) => {
+                    // Set the Cache-Control header for static content.
+                    var headers = new ResponseHeaders(context.Context.Response.Headers);
+
+                    // All static content is cacheable by intermediates.
+                    headers.CacheControl = new CacheControlHeaderValue()
+                    {
+                        Public = true
+                    };
+
+                    if (headers.ContentType.IsSubsetOf(TextHtml))
+                    {
+                        // Don't set max-age for HTML files (e.g. /tzdb/). We'll just send an ETag and let the browser
+                        // figure out when to revalidate.
+                        return;
+                    }
+
+                    // Otherwise send a sensible max-age value. If the request was made with a file version query
+                    // parameter (?v=..., as sent by asp-append-version=true), then cache the response 'indefinitely'.
+                    // Otherwise (e.g. /favicon.ico, /fonts/, /robots.txt), cache for a day.
+                    if (context.Context.Request.Query.ContainsKey("v"))
+                    {
+                        headers.CacheControl = new CacheControlHeaderValue()
+                        {
+                            Public = true,
+                            MaxAge = TimeSpan.FromDays(365)
+                        };
+                    }
+                    else
+                    {
+                        headers.CacheControl = new CacheControlHeaderValue()
+                        {
+                            Public = true,
+                            MaxAge = TimeSpan.FromDays(1)
```

Comment:
Reduced to a few minutes in development mode. I think that's the best of both: it allows for iteration, but it's representative of the production mode too.